ocs_ci/ocs/cluster.py

```diff
@@ -1171,3 +1171,78 @@ def check_osd_tree_1az_aws(osd_tree, number_of_osds):
     all_hosts_flatten = [item for sublist in all_hosts for item in sublist]
 
     return check_osds_in_hosts_osd_tree(all_hosts_flatten, osd_tree)
+
+
+class CephClusterExternal(CephCluster):
+    """
+    Handle all external ceph cluster related functionalities
+    Assumption: Cephcluster Kind resource exists
+
+    """
+
+    def __init__(self):
+        self.POD = ocp.OCP(
+            kind='Pod', namespace=config.ENV_DATA['cluster_namespace']
+        )
+        self.CEPHCLUSTER = ocp.OCP(
+            kind='CephCluster', namespace=config.ENV_DATA['cluster_namespace']
+        )
+
+        self.wait_for_cluster_cr()
+        self._cluster_name = (
+            self.cluster_resource.get('metadata').get('name')
+        )
+        self._namespace = (
+            self.cluster_resource.get('metadata').get('namespace')
+        )
+        self.mcg_obj = MCG()
+
+        self.cluster = ocs.OCS(**self.cluster_resource)
+
+    @property
+    def cluster_name(self):
+        return self._cluster_name
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @retry(IndexError, 10, 3, 1)
+    def wait_for_cluster_cr(self):
+        """
+        we have to wait for cluster cr to appear else
+        it leads to list index out of range error
+
+        """
+        cluster_cr = self.CEPHCLUSTER.get()
+        self.cluster_resource = cluster_cr.get('items')[0]
+
+    def cluster_health_check(self, timeout=300):
+        """
+        This would be a comprehensive cluster health check
+        which includes checking pods, external ceph cluster health.
+        raise exceptions.CephHealthException("Cluster health is NOT OK")
+        """
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=3,
+            func=self.is_health_ok
+        )
```

Comment:
same. Do we want to fail if cephcluster is not in HEALTH_OK in oc get cephcluster ?

Reply:
@nehaberry Any concrete decision on what to do about HEALTH_OK ? as of now we would fail for any condition other than HEALTH_OK (based on test_deployment sanity checks).

Reply:
@nehaberry , Any updates on status and phase checks ? as of now we are still failing if HEALTH_OK is not met. Also test_deployment has default behavior same as other deployments, i.e as per ``` ocs_install_verification()``` which checks both StorageCluster Phase and HEALTH_OK output from toolbox pod. Let me know if this needs to be changed.

Reply:
Going as per existing cluster health validation